### PR TITLE
Add children prop to types

### LINKED ISSUE #25: `children` prop type for latest react
With React v18.2.0 I have error in TypeScript:
```
TS2322: Type '{ children: (false | Element | Element[] | (Element | null)[] | undefined)[]; targetBlockWidth: number; center: false; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<XMasonryProps, {}, any>> & Readonly<XMasonryProps>'.   Property 'children' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Component<XMasonryProps, {}, any>> & Readonly<XMasonryProps>'.
```

As I understand it means that in https://github.com/ZitRos/react-xmasonry/blob/master/types.d.ts it should have `children` property.

### PR DESCRIPTION
**Pull request by iofjuupasli**
Closes #25

### CODE REVIEW

**Comment by iofjuupasli** [comment]
https://stackoverflow.com/questions/71788254/react-18-typescript-children-fc/71809927#71809927

**Comment by iofjuupasli** [comment]
If you could publish it to npm right after the merge, I would be very grateful.

File: types.d.ts [hunk 1]
```diff
@@ -1,5 +1,6 @@
 declare module "react-xmasonry" {
     export interface XMasonryProps {
+      children?: React.ReactNode;
```
Comment (nikitaeverywhere) [inline types.d.ts:3]: Well, the only children of XMasonry component should be XBlock. Can we enforce it in the type?

Comment (iofjuupasli) [reply types.d.ts:3]: Can't make it work in my project. Tried:
```
  type XBlockElement =
    | React.ReactElement<XBlockProps>
    | boolean
    | null
    | undefined;
  export interface XMasonryProps {
    children?: XBlockElement | Array<XBlockElement>;
```

Getting `Type 'Element[]' is not assignable to type 'XBlockElement'`. Seems that `ComponentClass` doesn't pass prop types to render function.

Maybe let's just merge it as is, so at least it's compatible.

**Comment by nikitaeverywhere** [comment]
@iofjuupasli the changes are now published. Thanks for your contribution!

**Pull request merged**